lightningd/plugin.c

```diff
@@ -498,8 +498,14 @@ static struct io_plan *plugin_read_json(struct io_conn *conn,
 	       plugin->buffer + plugin->used, plugin->len_read);
 
 	/* Peek into the buffer increment we just read (including a tiny
-	 * overlap if we're not at the start) to see if we have a '\n\n'
-	 * separator. If not we can skip the JSON attempt. */
+	 * overlap if we're not at the start) to see if we have a '}'
+	 * separator. If not we can skip the JSON attempt.
+	 *
+	 * Our JSON parser is pretty good at incremental parsing, but
+	 * `getrawblock` gives a giant 2MB token, which forces it to re-parse
+	 * every time until we have all of it. However, we can't complete a
+	 * JSON object without a '}', so we do a cheaper check here.
+	*/
 	have_full = memmem(plugin->buffer + plugin->used - offset,
 			   plugin->len_read + offset, "}", 1);
```

Comment:
This can be memchr :)

And we no longer need offset.  I'll add a fixup (and seems like a file needs regen, too...)